### PR TITLE
Full post

### PR DESCRIPTION
**Pull request by MiyamotoAkira**
My views on REST

### CODE REVIEW

**Comment by MiyamotoAkira** [comment]
DON'T MERGE till I fix the date. This is just for review.

**Comment by scotteza** [comment]
Cool post! I enjoyed reading it.

There are some items that broke the flow for me as I read, so here is some resultant grammar/style nit-picking:

- You should do an end-to-end check on capitalizing things like "HTTP", "API" and "GraphQL" correctly.
- Remove the word "are" in the sentence "Having an API that uses http doesn't mean that you are have a REST Api".
- Maybe write "HATEOAS" as "HATEOAS (Hypermedia as the Engine of Application State)" in the "Elements of REST" list. This makes it clear that it relates to the "Rest is about hypermedia" section.
- Change "Could use the others as well" to "You could use the others as well".
- I don't understand this heading: "Swagger is not conductive of REST". Did you mean "Swagger is not conducive to REST"?

File: _posts/2018-09-03-rest-review.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+date: 2018-08-30 08:01:00
+description: Where I talk about what is, and what is not, the REST architecture.
+image: 
+    src: /assets/custom/img/blog/2018-08-13-automated-code-styling/caffeine-coffe-cup-6347.jpg
+tags: 
+- REST
+- Architecture
+---
+
+There are two events that have happened recently close to each other. The first one was a decision to actually learn GraphQl. I have been thinking about it for a while, but now is time to actually put the effort on it. The second one was a talk with my colleague [Luciano](https://codurance.com/publications/author/luciano-palma/) in which I incorrectly made a remark about resources on REST. Because of both (for the former as I want to compare, for the latter to assert my knowledge), I decided to re-read both the [Fielding PhD dissertation](https://www.ics.uci.edu/~fielding/pubs/dissertation/top.htm) and the book [Rest in Practice](http://restinpractice.com/book/) by Robinson, Webber and Parastatidis.
```
Comment (upsd) [inline _posts/2018-09-03-rest-review.md]: Do you mean `GraphQL` here?

File: _posts/2018-09-03-rest-review.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+One thing that I completely forgot is that he worked on quite a few projects like Http/1.0, Http/1.1, Apache, ... and that Http itself is based on REST (mostly, as he does talk about the definitely not RESTful cookies and the need to support non RESTful existing extensions).
+
+### Rest In Practice
+
+I think this was a fantastic book with a really good introduction to REST apis. When it came the time for me to work on my first Web Api, I read this book as it had all glowy recomendations. After reading it a second time a few years later, I can say that is still fresh and on point.
+
+## Web Api != REST Api
+
+Having an API that uses http doesn't mean that you are have a REST Api. I think this is an important distinction, as a lot of programmers I have seen do mix them. A Web API is an API that can be accessed trought the network. A REST api is an api that follows the REST architectural style.
```
Comment (upsd) [inline _posts/2018-09-03-rest-review.md]: Do you mean "can be accessed through the network"?

Comment (upsd) [inline _posts/2018-09-03-rest-review.md]: I would try to be consistent with your use of "API", there is a mixture of uppercase and lowercase.

**Comment by MiyamotoAkira** [comment]
All your suggestions acted upon, Scott

File: _posts/2018-09-03-rest-review.md (hunk 2 above)
Comment (MiyamotoAkira) [reply _posts/2018-09-03-rest-review.md]: Fixed

Comment (MiyamotoAkira) [reply _posts/2018-09-03-rest-review.md]: Fixed

File: _posts/2018-09-03-rest-review.md (hunk 1 above)
Comment (MiyamotoAkira) [reply _posts/2018-09-03-rest-review.md]: When I mention resources on REST? No, that one is fine.

Comment (upsd) [reply _posts/2018-09-03-rest-review.md]: Here: "The first one was a decision to actually learn GraphQl' -> "... GraphQL".

**Comment by MiyamotoAkira** [comment]
I just realized that I forgot the versioning section. Coming soon here

**Comment by MiyamotoAkira** [comment]
New versioning section available.

**Comment by richardjwild** [comment]
As HTTP is an initialism not a word then it should be HTTP, not Http. Also, I believe the correct case for ReST is to lowercase the e because it is not the first letter of a word.

**Comment by richardjwild** [comment]
glowy recomendations -> glowing recomendations.

**Comment by richardjwild** [comment]
But if your architecture, and your API doesn't obey the constraints and elements below it cannot be called a REST architecure/API

->

But if your architecture and your API don't obey the constraints and elements below, it cannot be called a REST architecture/API.

**Comment by richardjwild** [comment]
Colon needed at the end of "The six constrains indicated by Fielding are"

**Comment by richardjwild** [comment]
And colon needed at the end of "The four elements are indicated below"

**Comment by richardjwild** [comment]
The capitalisation on the headings are not consistent. E.g. there is:

*Richardson's maturity Levels*

and there is:

*Rest is about hypermedia*

and there is also:

*Architectural Style*

**Comment by richardjwild** [comment]
"you most certaintly are not going to end having a REST API"

certaintly -> certainly

**Pull request merged**